app/models/spree/payment_decorator.rb

```diff
@@ -30,15 +32,6 @@ def adjustment_label
       I18n.t('payment_method_fee')
     end
 
-    # This is called by the calculator of a payment method
-    def line_items
-      if order.complete? && Spree::Config[:track_inventory_levels]
-        order.line_items.select { |li| inventory_units.pluck(:variant_id).include?(li.variant_id) }
-      else
-        order.line_items
-      end
-    end
```

Comment:
but where is this check performed then? I just want to understand.

Reply:
@sauloperez Currently, stock is being checked upon `save` of the line item:

- Validate upon save: https://github.com/openfoodfoundation/spree/blob/step-6a/core/app/models/spree/line_item.rb#L15
- Original Spree implementation: [`Spree::LineItem#stock_availability`](https://github.com/openfoodfoundation/spree/blob/step-6a/core/app/models/spree/line_item.rb#L109) and [`Spree::LineItem#sufficient_stock?`](https://github.com/openfoodfoundation/spree/blob/step-6a/core/app/models/spree/line_item.rb#L61)
- OFN implementation: [`Spree::LineItem#stock_availability`](https://github.com/openfoodfoundation/openfoodnetwork/blob/4ed0831e4c540107d5b154e166f2753d2180fb75/app/models/spree/line_item_decorator.rb#L160) and [`Spree::LineItem#sufficient_stock?`](https://github.com/openfoodfoundation/openfoodnetwork/blob/4ed0831e4c540107d5b154e166f2753d2180fb75/app/models/spree/line_item_decorator.rb#L145)